rectangle.h

```diff
@@ -1,31 +1,37 @@
 #pragma once
 
-#include <glm/glm.hpp>
-
 #include "indexbuffer.h"
 #include "renderer.h"
 #include "shader.h"
 #include "texture.h"
 #include "vertexarray.h"
 #include "vertexbuffer.h"
 
+#include <glm/glm.hpp>
+
+#include <memory>
+
+static const unsigned int kIndicies[] = {
+    0, 1, 2,
+    2, 3, 0
+};
+
 class Rectangle
 {
 private:
-    float mWidth;
-    float mHeight;
-    VertexBuffer mVB;
-    VertexArray mVA;
-    IndexBuffer mIB;
-    Texture *mTexture;
+    float mWidth { 0.0f };
+    float mHeight { 0.0f };
+    VertexBuffer mVB { 4 * 4 * sizeof(float) };
+    VertexArray mVA {};
+    IndexBuffer mIB { kIndicies, 6 };
+    std::unique_ptr<Texture>mTexture { nullptr };
```

Comment:
lovely header initialization and unique pointer! Hope you see the usefulness soon enough :)